go-controller/pkg/util/gateway-init.go

```diff
@@ -172,46 +126,77 @@ func GatewayInit(clusterIPSubnet []string, systemID, nodeName, ifaceID, nicIP, n
 			"stderr: %q, error: %v", gatewayRouter, stdout, stderr, err)
 	}
 
+	_, joinSubnet, _ := net.ParseCIDR(joinSubnetStr)
+	prefixLen, _ := joinSubnet.Mask.Size()
+	gwLRPIp := NextIP(joinSubnet.IP)
+	drLRPIp := NextIP(gwLRPIp)
+	gwLRPMac := IPAddrToHWAddr(gwLRPIp)
+	drLRPMac := IPAddrToHWAddr(drLRPIp)
+
+	joinSwitch := "join_" + nodeName
+	// create the per-node join switch
+	stdout, stderr, err = RunOVNNbctl(
+		"--", "--may-exist", "ls-add", joinSwitch,
+		"--", "set", "logical_switch", joinSwitch, fmt.Sprintf("%s=%s", config.OtherConfigSubnet(), joinSubnetStr))
```

Comment:
@girishmg is it necessary to set other-config:<subnet> on the switch if we're statically allocating the IPs on it?

Reply:
@dcbw you are right. If we are statically allocating the IPs it's not required. 

However, if you remember there is one fix that we have which I have not merged upstream since we have not converged on the fix. The fix is for the hair-pin issue and in our case I have an extra pair of Local L3 Gateway and Local External Logical Switch. The Local L3 gateway hangs off of the per-node Join switch.